qa/integration-tests/src/test/java/io/camunda/it/migration/util/MigrationITInvocationProvider.java

```diff
@@ -0,0 +1,510 @@
+
+                  closeables.add(zeebeUtils.get(db).start86Broker(zeebeEnvironmentBefore.get(db)));
+                  zeebeClients.put(db, zeebeUtils.get(db).getZeebeClient());
+
+                  updateEnvMap(
+                      db,
+                      tasklistEnvironmentBefore,
+                      tasklistElasticsearchDefaultConfig(zeebeUtils.get(db)));
+                  var tasklistEs =
+                      tasklistUtils
+                          .get(db)
+                          .createTasklist(tasklistEnvironmentBefore.get(db), false);
+                  closeables.add(tasklistEs);
+
+                  if (beforeMigrationFunc != null) {
+                    beforeMigrationFunc.accept(zeebeClients.get(db), tasklistUtils.get(db));
+                  }
+
+                  /* Startup Sequence 8.7*/
+
+                  zeebeUtils.get(db).stop();
+                  tasklistUtils.get(db).stop();
+
+                  updateEnvMap(db, zeebeEnvironmentAfter, zeebe87ElasticsearchDefaultConfig());
+                  final var broker =
+                      zeebeUtils
+                          .get(db)
+                          .start87Broker(
+                              cfg -> {
+                                cfg.setClassName("io.camunda.exporter.CamundaExporter");
+                                cfg.setArgs(
+                                    Map.of(
+                                        "connect",
+                                        Map.of(
+                                            "url",
+                                            "http://"
+                                                + elasticsearchContainer.getHttpHostAddress()),
+                                        "bulk",
+                                        Map.of("size", 10, "delay", 1),
+                                        "index",
+                                        Map.of("shouldWaitForImporters", false)));
+                              },
+                              zeebeEnvironmentAfter.get(db));
+
+                  try {
+                    cloneProcesses(db);
+                  } catch (final IOException e) {
+                    throw new RuntimeException(e);
+                  }
+
+                  updateEnvMap(
+                      db,
+                      tasklistEnvironmentAfter,
+                      tasklistElasticsearchDefaultConfig(zeebeUtils.get(db)));
+                  tasklistEs =
+                      tasklistUtils.get(db).createTasklist(tasklistEnvironmentAfter.get(db), true);
+                  closeables.add(tasklistEs);
+
+                  testBrokers.put(db, broker);
+                  zeebeClients.put(db, zeebeUtils.get(db).getZeebeClient());
+
+                  if (afterMigrationFunc != null) {
+                    afterMigrationFunc.accept(zeebeClients.get(db), tasklistUtils.get(db));
+                  }
+
+                  closeables.add(broker);
+                  break;
+
```

Comment:
🔧 that is a lot of set up code. Can we reduce this, or make this more readable? Like introduce separate methods, also useful for reducing with the OS setup?